lib/rspec/matchers.rb

```diff
@@ -582,7 +582,7 @@ def eql(expected)
     #
     # @example
     #   expect(5).to       equal(5)   # Integers are equal
-    #   expect("5").not_to equal("5") # Strings that look the same are not the same object
```

Comment:
I believe there is no point to compare two string literals, so this comment describes the case when some non-literal actual value is compared with `be_equal` with an expected literal might result in a failure.

BTW, the use of `equal` instead of `be_equal` seems to be the real problem here.

Reply:
Sure.

```suggestion
    #   expect(+"5").not_to equal(+"5") # Strings that look the same are not the same object
```

So this is a way (or use `dup`). Is it better?

Reply:
What are you after with this change in the first place?

Reply:
https://github.com/rspec/rspec-expectations/pull/1303#discussion_r646165448 is the my motivation

Reply:
To me, the current doc says "don't use `equals` unless you absolutely know what you're doing, e.g. if you compare identical strings that are not the same object, the comparison would fail".
Do you think adding `+` would help to make this more clear for developers of all experience levels?

Reply:
Ah thanks, that is the documentation direction of this project, I understood.